src/Monolog/Formatter/MongoDBFormatter.php

```diff
@@ -100,6 +102,13 @@ protected function formatException(\Exception $exception, $nestingLevel)
 
     protected function formatDate(\DateTime $value, $nestingLevel)
     {
-        return new \MongoDate($value->getTimestamp());
+        $seconds = (int) $value->format('U');
+        $milliseconds = (int) $value->format('u') / 1000;
+
+        if ($seconds < 0) {
+            return new UTCDateTime($seconds * 1000 - $milliseconds);
+        } else {
+            return new UTCDateTime($seconds * 1000 + $milliseconds);
+        }
```

Comment:
@Seldaek: I'm not sure what the point of this class was, as it was never the default formatter for the MongoDB handler. I suppose people may be using it manually?

In any event, this update was required for the new driver's date class. This math is prone to causing integer overflow on 32-bit systems, but that shouldn't be an issue for PHP 7, since it supports 64-bit ints on all platforms.